src/components/Sidebar/shortcuts/useShortcuts.tsx

```diff
@@ -0,0 +1,49 @@
+import { useDevModeContext } from "@dashboard/components/DevModePanel/hooks";
+import { useNavigatorContext } from "@dashboard/components/Navigator/useNavigatorContext";
+import { Graphql } from "@dashboard/icons/Graphql";
+import { SearchIcon } from "@saleor/macaw-ui-next";
+import React, { useCallback, useMemo } from "react";
+import { useIntl } from "react-intl";
+
+import { shortcutsMessages } from "./messages";
+import { getControlKey } from "./utils";
+
+export const useShortcuts = () => {
+  const intl = useIntl();
+  const devContext = useDevModeContext();
+  const { setNavigatorVisibility } = useNavigatorContext();
+
+  const controlKey = getControlKey();
+
+  const handleOpenPlayground = useCallback(() => {
+    devContext.setDevModeContent("");
+    devContext.setVariables("");
+    devContext.setDevModeVisibility(true);
+  }, []);
+
+  const handleOpenSearch = useCallback(() => {
+    setNavigatorVisibility(true);
```

Comment:
the provider you created has its own state and you wrap the entire dashboard in it, so now when you make this call, it will re-render the whole application.

Both "navigator" and playground live within the app layout, so why not place them close to the logic that triggers them? Then you get rid of global context?